### ISSUE TITLE
Xcode 15: iOS, tvOS, watchOS(, visionOS) SDKs not available

### ISSUE THREAD

**Issue by stephencelis** [actions/runner-images#7766]
### Description

Xcode 15 support was rolled out with a big bug that prevents all non-macOS SDKs from being available. https://github.com/actions/runner-images/issues/7672 was closed, though @davdroman pointed to the likely fix [here](https://github.com/actions/runner-images/issues/7672#issuecomment-1599660753). @mikhailkoliada suggested filing a separate issue, so I've done so here.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [X] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

https://github.com/pointfreeco/swift-composable-architecture/actions/runs/5325873531/jobs/9647154495?pr=2214

### Is it regression?

Kinda. SDKs/simulators are downloaded on Xcode 14 but not 15

### Expected behavior

I expect to be able to run CI for an iOS/tvOS/watchOS library or application.

### Actual behavior

I cannot due to the SDKs not being downloaded.

### Repro steps

Run this CI:

https://github.com/pointfreeco/swift-composable-architecture/actions/runs/5325873531/jobs/9647154495?pr=2214

Note that it cannot find any iOS simulator.

**Comment by ilia-shipitsin** [comment]
@stephencelis , I agree that it looks as "regression" from your point of view.

actually, MacOS-13 was started from scratch using different approach, currently it is still beta.
I've changed issue to "feature request", it definitely makes sense to add those SDK.

**Comment by stephencelis** [comment]
@ilia-shipitsin macOS 13 doesn't have this problem with Xcode 14, though.

**Comment by alicerunsonfedora** [comment]
Foes #7769 also pull in the simulators for visionOS as well? I have a Swift package that I want to write a workflow for (that supports visionOS), but I'm receiving a fatalError because the SDK isn't available: https://github.com/alicerunsonfedora/fedigardens/actions/runs/5500854590/jobs/10024058562?pr=32

**Comment by zhuowei** [comment]
@alicerunsonfedora I don't think so: the macOS VM currently [only includes](https://github.com/actions/runner-images/blob/38e96ca31af53461098beda948145b79b23e60bc/images/macos/macos-13-Readme.md#xcode) Xcode 15 beta 1, which doesn't support building for visionOS. We'll have to wait until they upgrade to a newer Xcode beta.

**Comment by stephencelis** [comment]
@ilia-shipitsin An associated PR was merged 2 weeks ago but it looks like support for Xcode simulator support for iOS and other platforms is still unavailable.

**Comment by ilia-shipitsin** [comment]
@stephencelis , that is sad but true, we still run MacOS-13 dated `20230611.2`

![image](https://github.com/actions/runner-images/assets/125650415/dd8d4a29-67fd-4c57-83a9-00acae942469)

we caught some regression after mentioned PR merge, so we had to revert it and we did not deploy MacOS-13 because of that.
currently, I work on updating XCode-15 to beta4. It is still work in progress.

**Comment by Vyazovoy** [comment]
Hi, @ilia-shipitsin! Any updates about this issue? It would be awesome if you could provide some timeframe.

**Comment by ilia-shipitsin** [comment]
some news, we are unblocked in macos-13 image generation (we stuck 2 weeks after 13.4.1(c) release due to security changes).
I hope 2 weeks as top estimate

**Comment by ilia-shipitsin** [comment]
@stephencelis ,  we deployed new image today (macos-13.5 + xcode-15.0 Beta2), please try

**Comment by stephencelis** [comment]
@ilia-shipitsin It's getting further along but I am seeing crashes in XCTest that I haven't had time to investigate yet. It may be related to the specific beta version, though (Xcode 15 beta 5 is already out).

**Comment by ilia-shipitsin** [comment]
we observe `objc[13795]: Class DTXDecompressionException is implemented in both /System/Library/PrivateFrameworks/DTXConnectionServices.framework/Versions/A/DTXConnectionServices (0x1ee1048e0) and /Applications/Xcode-15.app/Contents/SharedFrameworks/DTXConnectionServices.framework/Versions/A/DTXConnectionServices (0x1042bf630). One of the two will be used. Which one is undefined.` bug on XCode 
15 beta 3 and above:

https://discuss.circleci.com/t/xcode-15-beta-4-released/48645#bug-known-bugs-beetle-2

we decided to stay on 15 Beta 2 until that is resolved

**Comment by ilia-shipitsin** [comment]
image was deployed, I'm closing this issue

**Comment by calda** [comment]
With the changes above, the visionOS SDK is still unavailable. Filed that issue here: https://github.com/actions/runner-images/issues/8144

**Comment by ilia-shipitsin** [comment]
we install SDKs as `xcodebuild -downloadAllPlatforms`, seems visionOS needs to be downloaded separately as `xcodebuild -downloadPlatform visionOS`
